obspy/core/trace.py

```diff
@@ -2298,7 +2298,13 @@ def _internal_add_processing_info(self, info):
         trace's :class:`~obspy.core.trace.Stats` object.
         """
         proc = self.stats.setdefault('processing', [])
-        proc.append(info)
+        if len(proc) == 99:
```

Comment:
I'd propose to add this as a class attribute on `Trace`, e.g. `_max_processing_info`. That way people are in full control of this and can set it to other values.